src/main/scala/streams/GameDef.scala

```diff
@@ -130,22 +130,24 @@ trait GameDef {
      * Returns the list of blocks that can be obtained by moving
      * the current block, together with the corresponding move.
      */
-    def neighbors: List[(Block, Move)] = ???
+    def neighbors: List[(Block, Move)] = {
+      List((left, Left), (right, Right), (up, Up), (down, Down))
+    }
 
     /**
      * Returns the list of positions reachable from the current block
      * which are inside the terrain.
      */
-    def legalNeighbors: List[(Block, Move)] = ???
+    def legalNeighbors: List[(Block, Move)] = neighbors.filter(pair => pair._1.isLegal)
```

Comment:
```suggestion
    def legalNeighbors: List[(Block, Move)] = neighbors.filter(_._1.isLegal)
```
Just another way, shorter :)